src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_ForEachStatement.cs

```diff
@@ -506,8 +501,8 @@ private BoundStatement RewriteStringForEachStatement(BoundForEachStatement node)
             if (deconstruction == null)
             {
                 // V v = /* expression */
-                iterationVarDecl = MakeLocalDeclaration(forEachSyntax, iterationVar, iterationVarValue);
-                iterationVariables = ImmutableArray.Create(iterationVar);
+                Debug.Assert(iterationVariables.Length == 1);
```

Comment:
Let's add a test for this scenario.

---
In reply to: [95262886](https://github.com/dotnet/roslyn/pull/16318#discussion_r95262886) [](ancestors = 95262886)